### ISSUE TITLE
Story: Publish and download core collections

### ISSUE THREAD

**Issue by chouseknecht** [ansible/galaxy_ng#171]
We need to test and document how the Tower team will use a local AH to programmatically publish and consume a set of core collections.

The Tower team is looking for a local AH to be its source for core collections needed to build an execution environment. That means that in a development environment or in a VM, a developer will need to run a script or playbook that uses `ansible-galaxy` to publish multiple collections to the local AH.

We need to develop, test and document a way to:

1. Obtain an API token programmatically
2. Create a namespace programmatically 
3. Configure the `ansible-galaxy` client
4. Publish multiple collections to the namespace via `ansible-galaxy`
5. Download collections via `ansible-galaxy`

**Comment by chouseknecht** [comment]
- [ ] Determine if we can obtain an API token programmatically. 
- [ ] Expose in v3 the namespace, token and any API endpoints required for programmatic access. Don't remove them from `_ui` until David signs off. 
- [ ] Determine if any auth changes/fixes are needed in `ansible-galaxy`
- [ ] Document the workflow from end-to-end

**Comment by alikins** [comment]
@chouseknecht For #1 in https://github.com/ansible/galaxy_ng/issues/171#issue-619121672

"Obtain an API token programmatically"  - 

1. Is the Token in question the SSO/keycloak like is used in cloud.redhat.com for ansible-galaxy authentication via it's server 'auth_url' and 'token'? This can be done 'programmaticly' in the sense that Tower can get a SSO token from the 'auth_url' if needed.  

and/or

2. The django session token as provided at the /_ui/auth/{login,token} endpoints. This should be able to be done programmatically, but would required use of credentials stored in tower

**Comment by chouseknecht** [comment]
@alikins 

1. The token in question is a DRF token, not an SSO token 
2. We will not use the `ansible-galaxy` client to obtain the token
3. @cutwater will create a new v3 endpoint for obtaining a token
4. This new endpoint will not be accessible on c.rh.c

**Comment by alikins** [comment]
"Obtain an API token programmatically" 

ie, enough scriptable info to get a token that can be configured in ansible-galaxy config.
Possibly sub in. 

Critical step is validating ansible-galaxy can use a drf token granted from automation-hub.

Note drf token is not bearer token, so there is a reasonable chance that the ansible-galaxy client make not be able to make use of it for v3.

Need:
- current status of ansible-galaxy
- current status of galaxy_ng drf token acquisition
  -  at this moment, exists a /v3/_ui/auth/token/  but sounds like it is likely to get a new endpoint
      for getting django/drf token directly via basic auth.
        - the /auth/ endpoints may move as a whole to somewhere "public" under /v3/ (ala, https://github.com/ansible/galaxy_ng/issues/181)
- potential changes required in ansible-galaxy
- potential changes required in galaxy_ng

Basically, what needs to happen to allow ansible-galaxy to authenticate to galaxy_ng standalone with drf style tokens.

**Comment by chouseknecht** [comment]
Documentation on automated collection upload is here: https://github.com/ansible/galaxy_ng/wiki/Automating-Collection-Uploads

We've proven that using the end-user install to setup a Galaxy server, we can programmatically obtain a token, create a namespace, publish a collection, and download a collection.
